### DISCUSSION TITLE
Roles and role separations, profile notes.

### DISCUSSION THREAD

**Discussion by AlexanderRehm** [stoatchat/discussions#197] [Feature Suggestions]
Here's a suggestions for roles, role visual separations and notes for profiles

![image (1)](https://user-images.githubusercontent.com/93271971/139092893-294db32d-1cf2-41d3-af2f-8dc156a9b801.png)
Only notes section will be seen when user is looking at someone's profile in DM or chat group and other places except servers

![image](https://user-images.githubusercontent.com/93271971/139092902-8be8588e-03e6-43a4-823e-70b0980e8c5c.png)
![image (2)](https://user-images.githubusercontent.com/93271971/139092907-365682be-0879-4faf-92ad-4ef4d959827d.png)

**Comment by Pukimaa** [comment]
This would be great. Would allow for stuff like Pronoun Roles.

**Reply by Rexogamer** [reply]
These hopefully won't be necessary as we plan on adding built in support for pronouns at some point

**Comment by nulldg** [comment]
this would be a great opportunity to make special fields (such as pronouns) machine-readable. fields like these would be primarily useful to bots who can't easily read bios for user-generated metadata.

this is something skype had implemented well. in comparison, the approach discord seems to be interested in taking (ie just adding a single textbox without validation) makes it rather useless.
just like skype, it should be reliably organized and validated (eg a combobox for pronoun sets and a validator for custom pronouns, a colour picker for favourite colour, dropdown list of countries for country, dropdown list of timezones/offsets for timezone, etc etc), because if not, then we might as well just keep this simple and keep it in the user bio, letting the user present it their own way.

**Reply by Rexogamer** [reply]
I don't quite agree with the "validation or not at all" argument - for example, if you force pronouns to be put in the user bio you can't feasibly show them in the actual channel view; however, validation isn't a bad idea (irt pronouns I'd skip the preselected options and suggest just having a validated text field; for the others dropdowns/pickers would work well)

**Reply by nulldg** [reply]
the wonderful thing about a combobox is that it gives you examples (`he/him/his`, `she/her/her`, `they/them/their`, `it/it/its`) which you can select from (in the pronoun case, these four are extraordinarily common), and if the list doesn't contain the ones you want, you can also type in your own custom pronouns directly into the box. the validator would simply require that it follow the same format, which would be enough to allow bots/libraries to parse it. it's like a hybrid between a textbox and a list, offering the best of both.
the list helps imply what the format should look like, utilizing the user's intuition of first/second/third person rather than expecting explicit knowledge of it.

**Reply by Rexogamer** [reply]
That mostly seems fine but my main concern here is localisation - while pronouns as a concept are pretty universal and custom pronouns will cover any potential set, specific *sets* often aren't (for example, in German there isn't really a great equivalent of they/them) so I'd be somewhat hesitant to try and include default sets/imposing English pronouns on users of other languages isn't the best idea; however, giving the user some kind of hint would be useful and we *could* have localised arrays (e.g. he/she/they/it for English, der/die/das for German) although that would require some work

**Reply by nulldg** [reply]
that's a valid point. i'd argue it's necessary to make it localized if you want a bot/UI to refer to someone in a different language to their pronouns, otherwise we'd occasionally end up with silly situations like a string saying "[german user] changed **ihr** nickname".
custom pronouns would simply not be localized since the grammatical gender is unknown, and text written in languages without pronouns will either default to whatever the culture's preferred neuter pronouns are, or simply not use them, depending on how the language works.
